ndsl/dsl/typing.py

```diff
@@ -63,10 +63,19 @@ def global_set_floating_point_precision():
 IntFieldIJ = Field[gtscript.IJ, Int]
 IntFieldK = Field[gtscript.K, Int]
 BoolField = Field[gtscript.IJK, Bool]
+BoolFieldIJ = Field[gtscript.IJ, Bool]
 
 Index3D = Tuple[int, int, int]
 
 
+def set_4d_field_size(n, dtype):
```

Comment:
Ping @CharlesKrop, we have a helper function in NDSL to get 4D field types after this PR is merged

Reply:
for 4D array, do we still keep k as vertical direction? so the verticalLoop node is the same as before, and horizontal execution nodes contain 3d loops?